### ISSUE TITLE
A manifest constant written with the wrong keyword crashes the compiler

### ISSUE THREAD

**Issue by dukc** [bridgedotnet/Bridge#3711]
### Steps To Reproduce

Write

```d
private enum int colourPaletteHeight = 0x18;
```

This can happen if you are used to D language when intending to write:

```c#
private const int colourPaletteHeight = 0x18;
```

### Expected Result

```
   Bridge.NET Compiler error: exception level: 0 - C# Compilation Failed
        (###,##): CS####: "int" is not a valid enum identifier because it is a reserved keyword
```

(Or something like that)

### Actual Result

```
    Bridge.NET Compiler error: exception level: 0 - Syötemerkkijonon muoto ei ollut kelvollinen.
Stack trace:
   kohteessa System.Text.StringBuilder.FormatError()
   kohteessa System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   kohteessa System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   kohteessa System.String.Format(String format, Object[] args)
   kohteessa Bridge.Translator.TranslatorException.Throw(String format, Object[] args)
   kohteessa Bridge.Translator.Translator.BuildAssembly()
   kohteessa Bridge.Translator.Translator.Translate()
   kohteessa CallSite.Target(Closure , CallSite , Object )
   kohteessa System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid1[T0](CallSite site, T0 arg0)
   kohteessa Bridge.CLI.Program.Main(String[] args)
```

"Syötemerkkijononjonon muoto ei ollut kelvollinen" == "Input string was not of valid form"

**Comment by fabriciomurta** [comment]
Thanks for reporting this crash! Just to confirm, in the end it just turns an unsuccessful build into a failed build, right? I mean, build-wise, it should fail no matter what.

**Comment by vladsch** [comment]
It is already fixed in 17.3.0

**Comment by fabriciomurta** [comment]
Being realistic, this does not necessarily means any syntax error won't make the compiler to crash. If you see a situation where it still does, please don't hesitate on reporting to us!
